### ISSUE TITLE
Bold Italic?

### ISSUE THREAD

**Issue by matype** [0xType/0xProto#85]
## Why

- to complete the RIBBI set
- Shell-prompts often use it?

https://github.com/0xType/0xProto/issues/82#issuecomment-2324305929

## Why not now

https://github.com/0xType/0xProto/issues/79#issuecomment-2283765005

> By the way, I will not be making a Bold Italic for 0xProto.
>
> Because 0xProto is a programming font, and while Bold or Italic is necessary as a means of highlighting other than by color, wouldn't a Bold and Italic family be fine without it?
>
> I know that it is not a problem to have. But I have a lot of other projects I want to do besides 0xProto, and I don't want to add as many things as possible that I need to keep maintaining.

It is possible that I am not aware of the seriousness of the Bold Italic use case.

**Comment by Finii** [comment]
As the fist why is from me...

I tried to prove the second bullet point, but I could not find any screenshot where a prompt uses both bold and italic in parallel. So probably that does not exist (too often). I think I remember two-line prompts where some information is highlighted by BOLD and some other by ITALIC. Together with color.

The I tried to edit my Powerline setup to show some BoldItalic stuff, but while some in bold and some in italic works it does not render `{"italic", "bold"}` :thinking: 

![image](https://github.com/user-attachments/assets/3ad5627a-35db-463c-914a-82e8b2aab1a1)
_My powerline colorscheme default file_

And then, the terminal is controlled by ANSI sequences, and there just is no BoldItalic sequence:

![image](https://github.com/user-attachments/assets/013ce297-b604-4a75-baf5-8fb5dc056273)

![image](https://github.com/user-attachments/assets/99beab3b-f0b9-4aa1-8242-7cc73ecf055f)

_https://stackoverflow.com/questions/4842424/list-of-ansi-color-escape-sequences_

### Conclusion

> It is possible that I am not aware of the seriousness of the Bold Italic use case.

No, you are right and I was mislead. Sorry for the red herring! Ordinary fonts usually come in RIBBIs,, but for terminals there seems to be no use-case.

Sorry, Fini

**Comment by mcchrish** [comment]
Bold italic is certainly allowed in terminals.
```
echo -e "\e[3m\e[1mbold italic\e[0m"
```
<img width="288" alt="image" src="https://github.com/user-attachments/assets/8516a9cf-792b-43b1-8fc6-9654b7fcee2c">

But the absence of it is not much felt to be honest. I think editor themes rarely make use it, while the most common use case that I have seen is in comment docs like:
<img width="181" alt="image" src="https://github.com/user-attachments/assets/c8893de4-2092-45c5-b959-2efe9441657d">
Where it blends with italic comments and add emphasis on doc tags.

**Comment by matype** [comment]
OK, I won't start building the Bold Italic right away since it doesn't seem urgent or important. I won't close this issue, but I'll leave it as a task to work on gradually when I have time 🙂

**Comment by Finii** [comment]
@mcchrish 

Ah, right (looking surprised). Thanks.
For me this shows it a bit better / more obvious:

![image](https://github.com/user-attachments/assets/6f8988f1-51d6-497f-980e-a65c0b9ca0db)

I have also seen this cursive comment with bold, but never when I work. Is that not only in some IDE like VSCode or something? I have never seen it in a terminal, I believe. But then, I never do JS, except when it is unavoidable. Maybe even neovim does that (trying)... No. But it does show a tiny JS icon :laughing: 

![image](https://github.com/user-attachments/assets/91f7098a-2ebf-43ed-9d89-feac5c9ffea5)

**Comment by AndydeCleyre** [comment]
Bold Italic *isn't* implemented? So is wezterm or underlying font stuff faking it on the fly? Looks good to me:

(Zsh)

```console
$ print -lP 'Regular:     eizrm' '%BBold:        eizrm%b' '\e[3mItalic:      eizrm\e[0m' '\e[3m%BBold Italic: eizrm%b\e[0m'
```

![image](https://github.com/user-attachments/assets/a5bc5fa8-20f9-485c-bf13-4fea94f88774)

Though now that I stare at this screenshot I do wonder about the `ol` kerning in `Bold`.

**Comment by matype** [comment]
I'm not really sure how the non-existent Bold Italic is completed or how it works 🤔 

> Though now that I stare at this screenshot I do wonder about the `ol` kerning in `Bold`.

@AndydeCleyre Exactly, but given that 0xProto is monospaced, each character must fit within the same width box. As long as it is a monospaced font, there will be both high-density and low-density glyphs in the box.

**Comment by mserajnik** [comment]
> Bold Italic _isn't_ implemented? So is wezterm or underlying font stuff faking it on the fly? Looks good to me:
> 
> (Zsh)
> 
> ```
> $ print -lP 'Regular:     eizrm' '%BBold:        eizrm%b' '\e[3mItalic:      eizrm\e[0m' '\e[3m%BBold Italic: eizrm%b\e[0m'
> ```
> 
> ![image](https://private-user-images.githubusercontent.com/1787385/377953311-a5bc5fa8-20f9-485c-bf13-4fea94f88774.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzAyOTE0MDEsIm5iZiI6MTczMDI5MTEwMSwicGF0aCI6Ii8xNzg3Mzg1LzM3Nzk1MzMxMS1hNWJjNWZhOC0yMGY5LTQ4NWMtYmYxMy00ZmVhOTRmODg3NzQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTAzMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEwMzBUMTIyNTAxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MWUwMTQ3YmI1MTdkM2Y4ZWYzNDgyNTVkM2I0ZTA2ZDE4ZjE5NTJmMThjZDk3ODk1MGFkYTRmY2I1ZDVmMDY2MyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.2AyT3D78Kvi0CpKzOy182rqSjZmSxn9KbGHJGGDrWZs)
> 
> Though now that I stare at this screenshot I do wonder about the `ol` kerning in `Bold`.

I can confirm this works in [kitty](https://sw.kovidgoyal.net/kitty/) as well (heavily zoomed in for legibility):
<img width="1582" alt="image" src="https://github.com/user-attachments/assets/c8fd41b9-15f2-40ec-bf20-212db3f8af74">
With the following configuration:
```ini
font_family family="0xProto"
bold_font Bold
italic_font Italic
bold_italic_font Bold Italic
```

**Comment by brzzdev** [comment]
Would still love to see Bold Italic added

**Comment by cos4ni2s** [comment]
Most of these quirks wouldn't exists if the variable technology was adopted.
The [reason](https://github.com/0xType/0xProto/issues/51#issuecomment-1871041738) behind not adopting it is slightly out of place.

**Comment by fefedimoraes** [comment]
I've just learned about 0xProto from the [**Trying to fine tune my font config - how to apply script variant?** post on Reddit](https://www.reddit.com/r/Ghostty/comments/1sx8f1f/trying_to_fine_tune_my_font_config_how_to_apply). The font looks awesome!

I am using it on Ghostty, and the only complaint the I have is the lack of the Bold Italic style. The reason is that whenever I have an active selection in Visual Mode &mdash; which uses Bold Italic &mdash; it falls back to the regular font with the ss01 feature disabled.

<img width="1032" height="90" alt="Image" src="https://github.com/user-attachments/assets/4e1de1cc-d88d-4774-bce4-b5b265db12b7" />

I have managed to land on a somewhat satisfactory result by forcing the Bold Italic style into just Italic. In Ghostty, this can be achieved with the following config:

```
font-style-bold-italic = Italic
```

<img width="1032" height="90" alt="Image" src="https://github.com/user-attachments/assets/e9f4cfba-7f15-423f-869d-4cd5a4469d70" />

While it looks good, it would be awesome to have Bold Italic being natively supported. Here are the results in Ghostty when using [the `print` command posted by @msaktype before](https://github.com/0xType/0xProto/issues/85#issuecomment-2446984879):

### No Italic override

<img width="280" height="121" alt="Image" src="https://github.com/user-attachments/assets/d61bb7ad-1ac3-4142-bbd3-87f09c7ad2ce" />

### With Italic override

<img width="280" height="121" alt="Image" src="https://github.com/user-attachments/assets/8a3cacae-9167-4c34-8388-fa44c88b341c" />

**Comment by arjpar** [comment]
I think if there was a bold italic master, you could generate variable fonts with the weight (wght) axis for Roman & Italic slants, using `fontmake`. I've already created a 0xProto Roman variable font with the wght axis, but it might be buggy (not all features are implemented). I agree with others and think having a bold italic font would be great if you have the time to make it. Great job on 0xProto!
